### PR TITLE
unit: check replicator listener events

### PR DESCRIPTION
**Pull request by jayahariv**
* when conflicts happens, make sure there is only single event is fired. 
* validated when exception happens and also when resolution succeeds. 

Ref: #2428

### CODE REVIEW

File: Objective-C/Tests/ReplicatorTest+CustomConflict.m [hunk 1]
```diff
@@ -233,6 +233,48 @@ - (void) testConflictResolverMergeDoc {
     AssertEqualObjects(savedDoc.toDictionary, exp);
 }
 
+- (void) testDocumentReplicationEventForConflictedDocs {
+    TestConflictResolver* resolver;
+    
+    // when resolution throws exception
```
Comment (snej) [inline Objective-C/Tests/ReplicatorTest+CustomConflict.m]: It doesn't throw an exception, it just returns nil...

Comment (jayahariv) [inline Objective-C/Tests/ReplicatorTest+CustomConflict.m]:
```suggestion
    // when document with wrong-id is returned, it will throw exception.
```

Comment (jayahariv) [reply Objective-C/Tests/ReplicatorTest+CustomConflict.m]: it throws an exception, when we supply a wrong doc id. updated the comment

File: Objective-C/Tests/ReplicatorTest+CustomConflict.m [hunk 2]
```diff
@@ -259,12 +259,55 @@ - (void) testConflictResolverMergeDoc {
...
+    [replicator removeChangeListenerWithToken: token];
+}
+
 - (void) testConflictResolverCalledTwice {
     NSString* docId = @"doc";
     NSDictionary* localData = @{@"key1": @"value1"};
     NSDictionary* remoteData = @{@"key2": @"value2"};
     [self makeConflictFor: docId withLocal: localData withRemote: remoteData];
     
+
```
Comment (jayahariv) [inline Objective-C/Tests/ReplicatorTest+CustomConflict.m]:
```suggestion
```

**Pull request merged**